File: services/unsplashService.py
import requests
import os
from dotenv import load_dotenv

# Load environment variables
load_dotenv()

class UnsplashService:
    def __init__(self):
        self.api_key = os.getenv('UNSPLASH_API_KEY')
        self.base_url = 'https://api.unsplash.com'
    
    def search_photos(self, query, per_page=10):
        """Search for photos related to a query"""
        try:
            url = f"{self.base_url}/search/photos"
            params = {
                'query': query,
                'per_page': per_page,
                'orientation': 'landscape'
            }
            headers = {
                'Authorization': f'Client-ID {self.api_key}'
            }
            
            response = requests.get(url, params=params, headers=headers)
            response.raise_for_status()
            
            data = response.json()
            photos = []
            
            for photo in data.get('results', []):
                photos.append({
                    'id': photo['id'],
                    'url': photo['urls']['regular'],
                    'thumb': photo['urls']['thumb'],
                    'description': photo['alt_description'],
                    'photographer': {
                        'name': photo['user']['name'],
                        'profile': photo['user']['links']['html']
                    }
                })
            
            return photos
        except requests.exceptions.RequestException as e:
            print(f"Error fetching photos: {e}")
            return None
        except KeyError as e:
            print(f"Error parsing photos data: {e}")
            return None
    
    def get_random_photo(self, query=None):
        """Get a random photo, optionally related to a query"""
        try:
            url = f"{self.base_url}/photos/random"
            params = {
                'orientation': 'landscape'
            }
            if query:
                params['query'] = query
            headers = {
                'Authorization': f'Client-ID {self.api_key}'
            }
            
            response = requests.get(url, params=params, headers=headers)
            response.raise_for_status()
            
            data = response.json()
            return {
                'id': data['id'],
                'url': data['urls']['regular'],
                'thumb': data['urls']['thumb'],
                'description': data['alt_description'],
                'photographer': {
                    'name': data['user']['name'],
                    'profile': data['user']['links']['html']
                }
            }
        except requests.exceptions.RequestException as e:
            print(f"Error fetching random photo: {e}")
            return None
        except KeyError as e:
            print(f"Error parsing random photo data: {e}")
            return None

# Example usage
if __name__ == "__main__":
    unsplash_service = UnsplashService()
    
    # Test searching for travel photos
    photos = unsplash_service.search_photos("travel", 5)
    if photos:
        print("Travel photos found:")
        for photo in photos:
            print(f"- {photo['description']} by {photo['photographer']['name']}")

File: services/unsplashService.js
// Unsplash service for travel images
const UNSPLASH_API_KEY = process.env.UNSPLASH_API_KEY;
const BASE_URL = 'https://api.unsplash.com';

class UnsplashService {
  // Search for travel-related photos
  static async searchPhotos(query, perPage = 10) {
    try {
      const response = await fetch(
        `${BASE_URL}/search/photos?query=${encodeURIComponent(query)}&per_page=${perPage}&orientation=landscape`,
        {
          headers: {
            'Authorization': `Client-ID ${UNSPLASH_API_KEY}`
          }
        }
      );
      
      if (!response.ok) {
        throw new Error(`Unsplash API error: ${response.status}`);
      }
      
      const data = await response.json();
      return data.results.map(photo => ({
        id: photo.id,
        url: photo.urls.regular,
        thumb: photo.urls.thumb,
        description: photo.alt_description,
        photographer: {
          name: photo.user.name,
          profile: photo.user.links.html
        }
      }));
    } catch (error) {
      console.error('Error fetching photos:', error);
      throw error;
    }
  }
  
  // Get a random travel photo
  static async getRandomPhoto() {
    try {
      const response = await fetch(
        `${BASE_URL}/photos/random?query=travel&orientation=landscape`,
        {
          headers: {
            'Authorization': `Client-ID ${UNSPLASH_API_KEY}`
          }
        }
      );
      
      if (!response.ok) {
        throw new Error(`Unsplash API error: ${response.status}`);
      }
      
      const data = await response.json();
      return {
        id: data.id,
        url: data.urls.regular,
        thumb: data.urls.thumb,
        description: data.alt_description,
        photographer: {
          name: data.user.name,
          profile: data.user.links.html
        }
      };
    } catch (error) {
      console.error('Error fetching random photo:', error);
      throw error;
    }
  }
}

module.exports = UnsplashService;